### ISSUE TITLE
cmake error

### ISSUE THREAD

**Issue by grants** [NuLL3rr0r/blog-subscription-service#1]
blog-subscription-service/CMakeModules/FindVMime.cmake:111: error: STRING sub-command REPLACE requires at least four arguments. dependencies.cmake:185 (FIND_PACKAGE) CMakeLists.txt:223 (INCLUDE)

**Comment by NuLL3rr0r** [comment]
Thank you so much for reporting this. I'll try to take a look at the end of this weekend and see what's wrong.

**Comment by NuLL3rr0r** [comment]
OK, I found some time and ran a quick test on FreeBSD and Funtoo Linux and couldn't reproduce the mentioned issue. Could you please elaborate more on this? Which OS or version of CMake are you using?

You could possibly change these two lines and run a test again:

```
STRING ( REPLACE "." ";" VMIME_VERSION_LIST ${VMIME_VERSION_STRING} )
STRING ( REPLACE "." ";" VMIME_API_LIST ${VMIME_API_STRING} )
```

To

```
STRING ( REPLACE "." ";" VMIME_VERSION_LIST "${VMIME_VERSION_STRING}" )
STRING ( REPLACE "." ";" VMIME_API_LIST "${VMIME_API_STRING}" )
```

I'm guessing ${VMIME_VERSION_STRING} and ${VMIME_API_STRING} are probably empty so instead of passing four arguments the script passes three.

**Comment by grants** [comment]
I tried again, and compiled successfully. Thx.

**Comment by NuLL3rr0r** [comment]
OK, indeed you've found a bug. I built VMime from https://github.com/kisli/vmime while a friend using my CMake files used VMime version 0.9.1. The main reason it happened older versions use API format 0:0:0 while the new version uses 1.0.0 format. So, I fixed the CMake file and now it works.

Here is the commit that fixes this issue: [819a4c2](https://github.com/NuLL3rr0r/blog-subscription-service/commit/819a4c2472f3dd3ceebff7a1de628f8e473ebb6e).
